server/app/routes.py

```diff
@@ -14,12 +15,20 @@ def hello_world():
 		return 'Hello World!'
 
 	@app.route('/videos_raw')
-	def videos_raw():
-		return jsonify(rss.raw_feed(app.config['watchlist']))
+	async def videos_raw():
+		feed = await asyncio.to_thread(lambda: rss.raw_feed(app.config['watchlist']))
+		return jsonify(feed)
 
 	@app.route('/videos')
-	def videos():
-		return jsonify(rss.summarize_watchlist(app.config['watchlist']))
+	async def videos():
+		summarize_func = lambda: rss.summarize_watchlist(app.config['watchlist'])
+		summary = await asyncio.to_thread(summarize_func)
+		return jsonify(summary)
+
+	async def validate_id(id):
+		url = f'https://www.youtube.com/channel/{id}'
+		info = await async_extract_info(url)
+		return info is None or 'uploader_id' not in info or info['uploader_id'] != id
```

Comment:
the function name and what it returns is very misleading. We can either keep the return statement and rename the function to `is_invalid_id` or negate the return statement and rename the function to `is_valid_id`